### ISSUE TITLE
On Distillation of Guided Diffusion Models

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#187]
# TL; DR

- Introduce 2-step distillation approach to relieve the computational burden of classifier-free guidance diffusion models.

# Motivation

- Classifier-free diffusion models demonstrate impressive fidelity-diversity trade-off, but requires double inference for conditional sampling.

# Formulation

### Step 1: Merge two models into one

- Train a student model to imitate the outputs of the teacher model (conditional + unconditional) for all time steps.
- This model is conditioned on the guidance weight *w*.

### Step 2: Progressively cut the sampling steps in half

- Train a second student model that progressively imitates the first student model outputs with halving sampling steps. (Bootstrapped)

# Evaluation

![image](https://user-images.githubusercontent.com/46308407/201329821-ac817040-b8ab-4c9a-8a19-7e17fb23dd5c.png)
![image](https://user-images.githubusercontent.com/46308407/201329846-14254228-3b62-4638-b2ab-24711a31833f.png)
![image](https://user-images.githubusercontent.com/46308407/201329860-7ef22ce8-ae28-4481-98a0-9c64eda23af4.png)
![image](https://user-images.githubusercontent.com/46308407/201329879-466b4888-69a4-4341-a613-1f49c0ce3144.png)

- The domain transfer is done by dual implicit bridges. That is, encoding (forward DDIM) with one diffuser and decoding (denoising) with another.

# Inspiration

- One of the major bottlenecks in diffusion models was their heavy inference cost. Now that progressive distillation can be applied to classifier-free diffusion models, this problem is pretty much solved.
